### PR TITLE
feat(DatePicker): DatePicker Improvements

### LINKED ISSUE #1259: DatePicker - Keyboard control

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #1259

## Overview

Refactor DatePicker to improve selection behaviour and add keyboard control.

## Reason

>There are some strange edge case behaviours and we need keyboard control to improve accessibility.

## Work carried out

- [x] Major refactor of picker implementation
- [x] Implement keyboard control
- [x] Clean up automated tests

## Screenshots

Screenshot 2020-08-04 at 12 18 59
Screenshot 2020-08-04 at 12 19 18

### CODE REVIEW

File: packages/css-framework/src/components/_date-picker.scss [hunk 1]
```diff
@@ -151,42 +151,37 @@
   }
```
Comment (thyhjwb6) [inline packages/css-framework/src/components/_date-picker.scss:155]: There are a few places where `f.spacing` could be used here.

File: packages/react-component-library/package.json [hunk 2]
```diff
@@ -136,17 +136,18 @@
   ],
   "dependencies": {
     "@datepicker-react/hooks": "^2.4.1",
-    "@royalnavy/css-framework": "^2.13.0-alpha.3",
-    "@royalnavy/eslint-config-react": "^2.13.0-alpha.3",
-    "@royalnavy/fonts": "^2.13.0-alpha.3",
-    "@royalnavy/icon-library": "^2.13.0-alpha.3",
+    "@royalnavy/css-framework": "^2.12.1-alpha.4",
```
Comment (thyhjwb6) [inline packages/react-component-library/package.json]: Bad merge?

File: packages/react-component-library/src/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -79,49 +63,28 @@ export const DatePicker: React.FC<DatePickerProps> = ({
...
+  function handleDayClick(day: Date) {
+    const newState = isRange
+      ? (DateUtils.addDayToRange(day, state as RangeModifier) as StateObject)
+      : ({ from: day, to: day } as StateObject)
+
+    setState(newState)
 
     if (onChange) {
-      onChange(data)
+      onChange(newState)
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/DatePicker/DatePicker.tsx]: This is a breaking change..?

File: packages/react-component-library/src/components/DatePicker/DatePickerInput.tsx [hunk 4]
```diff
@@ -63,6 +63,7 @@ export const DatePickerInput = forwardRef(
             type="button"
             className="rn-date-picker__end-adornment"
             onClick={isOpen ? onClose : onFocus}
+            disabled={isDisabled}
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/DatePicker/DatePickerInput.tsx:66]: Could there have been a failing test for this?

**Review by thyhjwb6 — CHANGES_REQUESTED** [review]
Looks good.. a few comments and a concern around this being a breaking change.

File: packages/react-component-library/package.json (hunk 2 above)
Comment (m7kvqbe1) [reply packages/react-component-library/package.json]: Good spot

File: packages/react-component-library/src/components/DatePicker/DatePicker.tsx (hunk 3 above)
Comment (m7kvqbe1) [reply packages/react-component-library/src/components/DatePicker/DatePicker.tsx]: This has been updated so that the old callback signature is now retained.

File: packages/css-framework/src/components/_date-picker.scss (hunk 1 above)
Comment (m7kvqbe1) [reply packages/css-framework/src/components/_date-picker.scss:155]: Could you further clarify where else could utilise `f.spacing`?

Comment (thyhjwb6) [reply packages/css-framework/src/components/_date-picker.scss:155]: If you are happy with it then I am happy too 👍

**Review by thyhjwb6 — APPROVED** [review]
Looks good, nice updates 👍
